### PR TITLE
[EI-TOOL-73] Add a / by default to an API context

### PR DESCRIPTION
**Pull request by Bhashinee**
Fix the issue https://github.com/wso2/devstudio-tooling-ei/issues/73

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 1]
```diff
@@ -144,36 +144,40 @@ protected boolean isRequiredWorkingSet() {
...
-		content += "\" name=\"";
-		content += artifactModel.getName();
-		content += "\""; 
-		if(artifactModel.getHostname()!=null && artifactModel.getHostname().length()>0){
-			content += " hostname=\"";
-			content += artifactModel.getHostname();
-			content += "\"";
+		content.append("<api xmlns=\"");
+		content.append("http://ws.apache.org/ns/synapse\"");
+		content.append(" context=\"");
```
Comment (milindaperera) [inline plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Why do we use multiple appends here? we can do it single append

File: plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 2]
```diff
@@ -144,36 +144,40 @@ protected boolean isRequiredWorkingSet() {
...
+			content.append(artifactModel.getPort());
+			content.append("\"");
+		}
+		content.append(">\n");
+		content.append("<resource methods=\"GET\">");
+		content.append("<inSequence/>");
+		content.append("<outSequence/>");
+		content.append("<faultSequence/>");
+		content.append("</resource>\n");
+		content.append("</api>");
```
Comment (milindaperera) [inline plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: These all appends can be merged to one with one big string

File: plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 3]
```diff
@@ -144,36 +144,40 @@ protected boolean isRequiredWorkingSet() {
...
-			content += " hostname=\"";
-			content += artifactModel.getHostname();
-			content += "\"";
+		content.append("<api xmlns=\"");
+		content.append("http://ws.apache.org/ns/synapse\"");
+		content.append(" context=\"");
+		if (!artifactModel.getContext().startsWith("/")) {
+			content.append("/").append (artifactModel.getContext());
+		} else {
+			content.append(artifactModel.getContext());
```
Comment (manoramahp) [inline plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Shall we re-arrange the if-else block as below,
if (artifactModel.getContext().startsWith("/")) {
 	content.append(artifactModel.getContext());
} else {
 	content.append("/").append (artifactModel.getContext());
}

Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Done

File: plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java (hunk 2 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Done

File: plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java (hunk 1 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Done

File: plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 4]
```diff
@@ -144,36 +144,26 @@ protected boolean isRequiredWorkingSet() {
...
-		content += "\" name=\"";
-		content += artifactModel.getName();
-		content += "\""; 
-		if(artifactModel.getHostname()!=null && artifactModel.getHostname().length()>0){
-			content += " hostname=\"";
-			content += artifactModel.getHostname();
-			content += "\"";
+		content.append("<api xmlns=\"http://ws.apache.org/ns/synapse\" context=\"");
+		if (artifactModel.getContext().startsWith("/")) {
+			content.append(artifactModel.getContext());
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Shall we correct the indentation as per wso2 standard?

Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Fixed in this PR : https://github.com/wso2/devstudio-tooling-esb/pull/272

File: plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 5]
```diff
@@ -144,36 +144,26 @@ protected boolean isRequiredWorkingSet() {
...
+		} else {
+			content.append("/").append(artifactModel.getContext());
```
Comment (Bhashinee) [inline plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Fixed in this PR : #272

File: plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 6]
```diff
@@ -144,36 +144,26 @@ protected boolean isRequiredWorkingSet() {
...
 		}
-		if(artifactModel.getPort()>0){
-			content += " port=\"";
-			content += artifactModel.getPort();
-			content += "\"";
+		content.append("\" name=\"").append(artifactModel.getName()).append("\"");
+		if (artifactModel.getHostname() != null && artifactModel.getHostname().length() > 0) {
+			content.append(" hostname=\"").append(artifactModel.getHostname()).append("\"");
```
Comment (Bhashinee) [inline plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Fixed in this PR : #272

File: plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 7]
```diff
@@ -144,36 +144,26 @@ protected boolean isRequiredWorkingSet() {
...
-		content += ">\n";
-		content += "<resource methods=\"GET\">";
-		content += "<inSequence/>";
-		content += "<outSequence/>";
-		content += "<faultSequence/>";
-		content += "</resource>\n";
-		content += "</api>";
-		return content;
+		if (artifactModel.getPort() > 0) {
+			content.append(" port=\"").append(artifactModel.getPort()).append("\"");
```
Comment (Bhashinee) [inline plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java]: Fixed in this PR : #272

**Pull request merged**